src/server/json_family_test.cc

```diff
@@ -1400,6 +1400,42 @@ TEST_F(JsonFamilyTest, Merge) {
   EXPECT_EQ(resp, "OK");
   resp = Run({"JSON.GET", "foo", "$"});
   EXPECT_EQ(resp, R"([{"common":4,"f2":2}])");
+
+  json = R"({
+  "ans": {
+    "x": {
+      "y" : {
+        "doubled": false,
+        "answers": [
+          "foo",
+          "bar"
+        ]
+      }
+    }
+  }
+  })";
+  resp = Run({"JSON.SET", "j2", "$", json});
+  ASSERT_EQ(resp, "OK");
+
+  patch = R"(
+    {"z": {
+      "doubled": false,
+      "answers": ["xxx",  "yyy"]
+     }
+     })";
+
+  resp = Run({"JSON.MERGE", "j2", "$.ans.x", patch});
+
+  EXPECT_EQ(resp, "OK");
+  resp = Run({"JSON.GET", "j2", "$"});
+  EXPECT_EQ(resp, R"([{"ans":{"x":{"y":{"answers":["foo","bar"],"doubled":false},)"
+                  R"("z":{"answers":["xxx","yyy"],"doubled":false}}}}])");
```

Comment:
Maybe it's worth adding to `patch` another value that updates one of the elements of the `y` element ?

Something like:
```
 R"( { "y": { "doubled": true } } )"
```

Such that the query is both an update to an existing element `y` and an insertion of the new element `z`